lib/api/src/identity/Identity.ts

```diff
@@ -0,0 +1,22 @@
+
+export abstract class Identity {
+
+  /**
+   * Generates a url to be used as authentication redirect url.
+   *
+   * @param path A pathname to append to the origin
+   * @return Authentication redirect url
+   */
+  public abstract getRedirectUrl (path: string) : string
```

Comment:
Would this benefit in some scenarios with `async`, i.e. `: Promise<string>`? Might allow the users to perform logic which depends on async methods.

Reply:
Actually since externato implements these APIs, this is probably not a valid point.

Reply:
I _think_ the only real use for this method is from within the subsequent `launchWebAuthFlow()`, which needs to append the redirect url to its configured authentication endpoint but if @kouks thinks this would be better implemented as async I'm happy to update

Reply:
Don't think we need to make it async... converting sync functions to promises is easy, the other way around not so much.